File: 100_same_tree/isSameTree.go
package sametree

//   Definition for a binary tree node.
type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}
 
func isSameTree(p *TreeNode, q *TreeNode) bool {
	type StackNode struct {
		n *TreeNode
		m *TreeNode
	}
	stack := []StackNode{
		{p, q},
	}

	for len(stack) > 0 {
		node := stack[len(stack)-1]
		stack = stack[:len(stack)-1]

		if node.n == nil && node.m == nil {
			continue
		} else if node.n == nil || node.m == nil {
			return false
		} else {
			if node.n.Val != node.m.Val {
				return false
			}

			stack = append(stack, StackNode{node.n.Right, node.m.Right})
			stack = append(stack, StackNode{node.n.Left, node.m.Left})
		}

	}
	
	return true
}

File: 100_same_tree/isSameTree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
        
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        stack = [(p, q)]

        while stack:
            n, m = stack.pop()

            if not n and not m:
                continue
            elif None in [n, m]:
                return False
            else:
                if n.val != m.val:
                    return False
                stack.append((n.right, m.right))
                stack.append((n.left, m.left))

        return True